_includes/js/thebelab.html

```diff
@@ -25,42 +24,45 @@
             }
 
             // Find all non-hardware, code cells, replace with Thebelab interactive code cells
-            const notHardwareCells = document.querySelectorAll('div.jb_cell:not(.tag_uses-hardware)')
-	    notHardwareCells.forEach((notHWCell, index) => {
-	    const codeCells = notHWCell.querySelectorAll('.input_area pre')
+            const hardwareTag = 'uses-hardware'
+            const hardwareQuery = `.jb_cell.tag_${hardwareTag}`
+            const noHardwareQuery = `.jb_cell:not(.tag_${hardwareTag})`
+            const codeCells = document.querySelectorAll(`${noHardwareQuery} pre`)
             codeCells.forEach((codeCell, index) => {
                 const id = codeCellId(index)
-                
-		// Clean up the language to make it work w/ CodeMirror and add it to the cell
+
+                // Clean up the language to make it work w/ CodeMirror and add it to the cell
                 dataLanguage = "{{ kernelName }}"
                 dataLanguage = detectLanguage(dataLanguage);
                 codeCell.setAttribute('data-language', dataLanguage)
                 codeCell.setAttribute('data-executable', 'true')
-		
+
                 // If the code cell is hidden, show it
                 var inputCheckbox = document.querySelector(`input#hidebtn${codeCell.id}`);
                 if (inputCheckbox !== null) {
                     setCodeCellVisibility(inputCheckbox, 'visible');
                 }
             });
-	    });
-
-	    // Add warning and link to IQX on uses-hardware cells
-	    const hardwareCells = document.querySelectorAll('.tag_uses-hardware')
-	    hardwareCells.forEach((hardwareCell, index) => {
-		    var iqxLink = window.location.pathname
-		    iqxLink = iqxLink.slice(9,-4)
-		    iqxLink += "ipynb"
-		    iqxLink = "{{ site.iqx_url }}/content" + iqxLink
-	
-		    var hardwareWarning = '<code>'
-		    hardwareWarning += '{{ site.hardware_warning_message }}'
-		    hardwareWarning += '<a href="' + iqxLink + '" target="_blank">'
-		    hardwareWarning += '{{ site.hardware_warning_linktxt }}'
-		    hardwareWarning += '</a></code>'
-
-		    hardwareCell.innerHTML = hardwareWarning + hardwareCell.innerHTML
-	    });
+
+
+            // Create link to IQX and hardware notice template
+            const currentPath = '{{ page.url }}'
+            const notebookPath = currentPath.replace(/\.html$/, '.ipynb')
+            const iqxUrl = `{{ site.iqx_url }}/content/${notebookPath}`
```

Comment:
This link is not directing me to the right place. I had the same problem: For some reason `page.url` always points to the first page you visited, and not the notebook page you're on. This is why I ended up using `window.location.pathname`.